frontend/src/app/question/question.component.ts

```diff
@@ -0,0 +1,241 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { ConfirmationService, MessageService } from 'primeng/api';
+import { ButtonModule } from 'primeng/button';
+import { ConfirmDialogModule } from 'primeng/confirmdialog';
+import { DialogModule } from 'primeng/dialog';
+import { DropdownModule } from 'primeng/dropdown';
+import { FileUploadModule } from 'primeng/fileupload';
+import { InputNumberModule } from 'primeng/inputnumber';
+import { InputTextModule } from 'primeng/inputtext';
+import { InputTextareaModule } from 'primeng/inputtextarea';
+import { RadioButtonModule } from 'primeng/radiobutton';
+import { RatingModule } from 'primeng/rating';
+import { RippleModule } from 'primeng/ripple';
+import { TableModule, TablePageEvent } from 'primeng/table';
+import { TagModule } from 'primeng/tag';
+import { ToastModule } from 'primeng/toast';
+import { ToolbarModule } from 'primeng/toolbar';
+import { Difficulty, DifficultyService, Question, QuestionService, Subject, SubjectService } from '../core/api';
+
+@Component({
+  selector: 'app-question',
+  standalone: true,
+  imports: [
+    TableModule, 
+    DialogModule, 
+    RippleModule, 
+    ButtonModule, 
+    ToastModule, 
+    ToolbarModule, 
+    ConfirmDialogModule, 
+    InputTextModule, 
+    InputTextareaModule, 
+    CommonModule, 
+    FileUploadModule, 
+    DropdownModule, 
+    TagModule, 
+    RadioButtonModule, 
+    RatingModule, 
+    InputTextModule, 
+    FormsModule, 
+    InputNumberModule,
+  ],
+  providers: [
+    MessageService, 
+    ConfirmationService, 
+    QuestionService,
+    SubjectService,
+    DifficultyService,
+  ],
+  templateUrl: './question.component.html',
+  styleUrl: './question.component.css'
+})
+export class QuestionComponent implements OnInit {
+  questionDialog: boolean = false;
+  questions!: Question[];
+  question!: Question;
+  selectedQuestions!: Question[] | null;
+
+  subjects!: Subject[];
+  subject!: Subject;
+
+  difficulties!: Difficulty[];
+  difficulty!: Difficulty;
+
+  submitted: boolean = false;
+  searchValue: string | undefined;
+
+
+  public count: number = 10;
+	public first: number = 0;
+	public rows: number = 10;
+
+  constructor(
+    private questionService: QuestionService, 
+    private messageService: MessageService, 
+    private confirmationService: ConfirmationService,
+    private subjectService: SubjectService,
+    private difficultyService: DifficultyService,
+  ) {}
+
+  public ngOnInit() {
+    this.updatePage();
+
+    this.subjectService.subjectList().subscribe(subjects => {
+      this.subjects = subjects.results.map(subject => {
```

Comment:
bruhhhhhhh, `this.subjects = subjects.results`, tại sao lại `map`

Reply:
![image](https://github.com/user-attachments/assets/22cb302b-6a88-4ee9-9b62-298c2b2d7ee8)
map để lấy subject.name